libs/validate_results.py

```diff
@@ -0,0 +1,112 @@
+import csv 
+import sys
+
+from libs.us_state_abbrev import us_state_abbrev
+
+"""
+A set of functions to validate the datasets we prepare
+"""
+
+OUTPUT_HEADERS_SHARE = [
+    'Province/State',
+    "Country/Region",
+    "Last Update",
+    "Latitude",
+    "Longitude",
+    "State/County FIPS Code",
+    'State Intervention',
+    '16d-HSPTLZD',
+    '32d-HSPTLZD',
+    '16d-LACKBEDS',
+    '32d-LACKBEDS',
+    'MEAN-HOSP',
+    'MEAN-DEATHS',
+    'PEAK-HOSP',
+    'PEAK-DEATHS',
+    "Current Deaths",
+    "Current Confirmed",
+    "Current Recovered",
+    "Current Active",
+    "Combined Key",
+    "County"
+]
+
+
+OUTPUT_HEADERS_STATES = OUTPUT_HEADERS_SHARE + []
+OUTPUT_HEADERS_COUNTIES = OUTPUT_HEADERS_SHARE + []
+
+EXPECTED_MISSING_STATES = set([
+    'Northern Mariana Islands', 'American Samoa', 'Virgin Islands', 'Puerto Rico', 'Guam'
+])
+
+EXPECTED_MISSING_STATES_FROM_COUNTES = set([
+    'District of Columbia'
+])
+
+class DataExportException(Exception): 
+    def __init__(self, key, message):
+        self.key = key
+        self.message = message
+
+def __raise_error_if_incorrect_headers(key, headers_list, df): 
```

Comment:
in general in python single underscores are the standard here.  Double underscores do some weird name mangling (though that might only be in classes) so they are usually shied away from.